homeassistant/components/airly/air_quality.py

```diff
@@ -0,0 +1,203 @@
+"""Support for the Airly service."""
+import asyncio
+import logging
+from datetime import timedelta
+
+import async_timeout
+from aiohttp.client_exceptions import ClientConnectorError
+from airly import Airly
+from airly.exceptions import AirlyError
+
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.components.air_quality import (
+    AirQualityEntity,
+    ATTR_AQI,
+    ATTR_PM_10,
+    ATTR_PM_2_5,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util import Throttle
+
+from .const import CONF_LANGUAGE, NO_AIRLY_SENSORS
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = "Data provided by Airly"
+
+ATTR_API_ADVICE = "ADVICE"
+ATTR_API_CAQI = "CAQI"
+ATTR_API_CAQI_DESCRIPTION = "DESCRIPTION"
+ATTR_API_CAQI_LEVEL = "LEVEL"
+ATTR_API_PM10 = "PM10"
+ATTR_API_PM10_LIMIT = "PM10_LIMIT"
+ATTR_API_PM10_PERCENT = "PM10_PERCENT"
+ATTR_API_PM25 = "PM25"
+ATTR_API_PM25_LIMIT = "PM25_LIMIT"
+ATTR_API_PM25_PERCENT = "PM25_PERCENT"
+
+LABEL_ADVICE = "advice"
+LABEL_AQI_LEVEL = f"{ATTR_AQI}_level"
+LABEL_PM_2_5_LIMIT = f"{ATTR_PM_2_5}_limit"
+LABEL_PM_2_5_PERCENT = f"{ATTR_PM_2_5}_percent_of_limit"
+LABEL_PM_10_LIMIT = f"{ATTR_PM_10}_limit"
+LABEL_PM_10_PERCENT = f"{ATTR_PM_10}_percent_of_limit"
+
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=10)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a Airly entities from a config_entry."""
+    api_key = config_entry.data[CONF_API_KEY]
+    name = config_entry.data[CONF_NAME]
+    latitude = config_entry.data[CONF_LATITUDE]
+    longitude = config_entry.data[CONF_LONGITUDE]
+    language = config_entry.data[CONF_LANGUAGE]
```

Comment:
We need to remove his option. I've checked with @balloob and we don't allow language options per integration. We don't have a solution for this kind of translation at the moment.

I'm sorry I didn't catch this earlier.

Reply:
There is no exception to this rule?

For example Dark Sky sensor has language as an option https://www.home-assistant.io/integrations/darksky/#language

Reply:
Or maybe integration could read user's language from hass?